datadog/resource_datadog_logs_pipelineorder.go

```diff
@@ -0,0 +1,78 @@
+package datadog
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/zorkian/go-datadog-api"
+)
+
+func resourceDatadogLogsPipelineOrder() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDatadogLogsPipelineOrderCreate,
+		Update: resourceDatadogLogsPipelineOrderUpdate,
+		Read:   resourceDatadogLogsPipelineOrderRead,
+		Delete: resourceDatadogLogsPipelineOrderDelete,
+		Exists: resourceDatadogLogsPipelineOrderExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {Type: schema.TypeString, Required: true},
+			"pipelines": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func resourceDatadogLogsPipelineOrderCreate(d *schema.ResourceData, meta interface{}) error {
+	return resourceDatadogLogsPipelineOrderUpdate(d, meta)
+}
+
+func resourceDatadogLogsPipelineOrderRead(d *schema.ResourceData, meta interface{}) error {
+	ddList, err := meta.(*datadog.Client).GetLogsPipelineList()
+	if err != nil {
+		return err
+	}
+
+	if err = d.Set("pipelines", ddList.PipelineIds); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceDatadogLogsPipelineOrderUpdate(d *schema.ResourceData, meta interface{}) error {
+	var ddPipelineList datadog.LogsPipelineList
+	tfList := d.Get("pipelines").([]interface{})
+	ddList := make([]string, len(tfList))
+	for i, id := range tfList {
+		ddList[i] = id.(string)
+	}
+	ddPipelineList.PipelineIds = ddList
+	var tfId string
+	if name, exists := d.GetOk("name"); exists {
+		tfId = name.(string)
+	}
+	if _, err := meta.(*datadog.Client).UpdateLogsPipelineList(&ddPipelineList); err != nil {
+		return fmt.Errorf("error updating logs pipeline list: (%s)", err.Error())
+	}
+	d.SetId(tfId)
+	return resourceDatadogLogsPipelineOrderRead(d, meta)
+}
+
+// The deletion of pipeline order is not supported from config API.
+// This function simply delete the pipeline order resource from terraform state.
+func resourceDatadogLogsPipelineOrderDelete(d *schema.ResourceData, meta interface{}) error {
+
+	return nil
+}
+
+func resourceDatadogLogsPipelineOrderExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	if pipelineList, err := meta.(*datadog.Client).GetLogsPipelineList(); err == nil && len(pipelineList.PipelineIds) > 0 {
```

Comment:
Several questions:
* Under what circumstances would the API return an error here?
* Why doesn't the order exist if there are 0 pipelines in it?

IIUC the pipeline order is a "singleton" resource that is always there, so this should always return true.

Reply:
> * Under what circumstances would the API return an error here?

API will not return any other errors than server error and network error. 

> * Why doesn't the order exist if there are 0 pipelines in it?

It’s a decision I made for returning `not exist` when list len is 0. I will remove this check as I think it further now, if we don’t delete, it makes sense to always exists.

> IIUC the pipeline order is a "singleton" resource that is always there, so this should always return true.

As discussed with @bkabrda I will always return `true, nil` for this function, as the server error and network error can be taking care of by other functions.